### ISSUE TITLE
Supporting `FeatureFlags` in `scaffolder` templates

### ISSUE THREAD

**Issue by benjdlambert** [backstage/backstage#8870]
## Feature Suggestion
When developing templates it would be really nice if we could show and hide certain parts of the form, either on a field level or a section level so that we can turn off experimental things for most users and turn them on to test with in a production environment.

## Possible Implementation
```yaml
apiVersion: scaffolder.backstage.io/v1beta3
kind: Template
metadata:
  name: test-v1beta3
  title: Test v1beta3
  description: Test V1 Beta 3 Demo Templates
spec:
  type: website
  owner: team-a
  parameters:
    - name: Enter some stuff
      description: Enter some stuff
      backstage:featureFlag: should-show-some-stuff
      properties:
        inputString:
          type: string
          title: string input test
        inputObject:
          type: object
          title: object input test
          description: a little nested thing never hurt anyone right?
          properties:
            first:
              type: string
              title: first
              backstage:featureFlag: should-show-some-stuff-first-option
            second:
              type: number
              title: second
  steps:
    - id: debug
      if: ${{ true === true }}
      name: Debug
      action: debug:log
      input:
        message: ${{ parameters.inputString }}
        extra: ${{ parameters.inputObject }}
```

Open to different options and bikeshedding the name a little more, but the big picture is here.

**Comment by benjdlambert** [comment]
I believe that this is closed by #9665, however, @djamaile is there relevant documentation for the feature?

**Comment by djamaile** [comment]
@benjdlambert ah completely forgot about the docs, I’ll do it this week!

**Comment by benjdlambert** [comment]
In the PR - just make sure to close this ticket in it, and then I think we can close this :)

**Comment by sabinasiddiqi** [comment]
How to we register feature flags that can be used in the scaffolder templates? And how do we assign them to specific users?

**Comment by benjdlambert** [comment]
You can use `useApi(featureFlagsApiRef)` to register flags, and then they will be in the `Feature Flag` menu under user settings. Or you can register them when creating a plugin yourself `createPlugin({ featureFlags: [] })` I believe.

**Comment by sabinasiddiqi** [comment]
Thank you! Is there a way for an admin user to assign flags instead of the user turning it on and off themselves?

Also there seems to be a UI bug in the user settings feature flags tab - I added a feature flag for the scaffolder plugin in Root.tsx like so:
```
  const featureFlagApi = useApi(featureFlagsApiRef);
  featureFlagApi.registerFlag({
    name: 'test-feature-flag',
    pluginId: 'scaffolder',
  });
```

And the first time I looked at feature flags in the user settings, it shows up once but when I click elsewhere and come back to user settings, the flag shows 10 times.
<img width="1473" alt="Screen Shot 2022-08-11 at 10 29 17 AM" src="https://user-images.githubusercontent.com/2702642/184196245-ad83dc54-f838-4ddd-8b1e-1d95a1c9189e.png">
